src/main/java/gregtech/common/items/tool/HarvestCropsBehavior.java

```diff
@@ -0,0 +1,90 @@
+package gregtech.common.items.tool;
+
+import com.google.common.collect.ImmutableSet;
+import gregtech.api.GTValues;
+import gregtech.api.items.toolitem.ToolHelper;
+import gregtech.api.items.toolitem.aoe.AoESymmetrical;
+import gregtech.api.items.toolitem.behavior.IToolBehavior;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockCrops;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.item.EntityItem;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Set;
+
+public class HarvestCropsBehavior implements IToolBehavior {
+
+    @Override
+    public void onBlockStartBreak(@Nonnull ItemStack stack, @Nonnull BlockPos pos, @Nonnull EntityPlayer player) {
+
+        AoESymmetrical aoeDefinition = ToolHelper.getAoEDefinition(stack);
+
+        Set<BlockPos> blocks;
+
+        if(aoeDefinition == AoESymmetrical.none()) {
+            blocks = ImmutableSet.of(pos);
+        } else {
+            Vec3d lookPos = player.getPositionEyes(1F);
+            Vec3d rotation = player.getLook(1);
+            Vec3d realLookPos = lookPos.add(rotation.x * 5, rotation.y * 5, rotation.z * 5);
+            RayTraceResult rayTraceResult = player.world.rayTraceBlocks(lookPos, realLookPos);
+
+            if (rayTraceResult == null) return;
+            if (rayTraceResult.typeOfHit != RayTraceResult.Type.BLOCK) return;
+            if (rayTraceResult.sideHit == null) return;
```

Comment:
This is duplicated in `GrassPathBehavior` and `HoeGroundBehavior` (but there it is returning `EnumActionResult`). This could be moved to function too with simple if here.